src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ArrayCreationAndInitializer.vb

```diff
@@ -0,0 +1,435 @@
+    Left: ILiteralExpression (Text: 0) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 0)
+    Right: ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  IArrayInitializer (OperationKind.ArrayInitializer)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of ArrayCreationExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+        <Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")>
+        Public Sub ArrayCreationOfImplicitlyTypedMultiDimensionalArrays_WithInitializer()
+            Dim source = <![CDATA[
+Class C
+    Public Sub F()
+        Dim a = {{{{1, 2}}}, {{{3, 4}}}}'BIND:"{{{{1, 2}}}, {{{3, 4}}}}"
+    End Sub
+End Class
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IArrayCreationExpression (Dimension sizes: 4, Element Type: System.Int32) (OperationKind.ArrayCreationExpression, Type: System.Int32(,,,))
+  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 2)
+  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 2)
+  IArrayInitializer (OperationKind.ArrayInitializer)
+    IArrayInitializer (OperationKind.ArrayInitializer)
+      IArrayInitializer (OperationKind.ArrayInitializer)
+        IArrayInitializer (OperationKind.ArrayInitializer)
+          ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+          ILiteralExpression (Text: 2) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 2)
+    IArrayInitializer (OperationKind.ArrayInitializer)
+      IArrayInitializer (OperationKind.ArrayInitializer)
+        IArrayInitializer (OperationKind.ArrayInitializer)
+          ILiteralExpression (Text: 3) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 3)
+          ILiteralExpression (Text: 4) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 4)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of CollectionInitializerSyntax)(source, expectedOperationTree)
+        End Sub
+
+        <Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")>
+        Public Sub ArrayCreationErrorCase_MissingDimension()
+            Dim source = <![CDATA[
+Class C
+    Public Sub F()
+        Dim a = New String(1,) {}'BIND:"New String(1,) {}"
+    End Sub
+End Class
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IArrayCreationExpression (Dimension sizes: 2, Element Type: System.String) (OperationKind.ArrayCreationExpression, Type: System.String(,), IsInvalid)
+  IBinaryOperatorExpression (BinaryOperationKind.IntegerAdd) (OperationKind.BinaryOperatorExpression, Type: System.Int32, Constant: 2)
```

Comment:
And again. In this case, the resulting IOp tree here is very different from the C# version. Is this another result of the highest/lowest difference?